crypto/rand/rand_lib.c

```diff
@@ -346,10 +349,12 @@ void rand_cleanup_int(void)
 {
     const RAND_METHOD *meth = default_RAND_meth;
 
+    rand_cleaning_up = 1;
+
     if (meth != NULL && meth->cleanup != NULL)
         meth->cleanup();
-    rand_pool_cleanup();
     RAND_set_rand_method(NULL);
+    rand_pool_cleanup();
```

Comment:
Swapping these two lines was my first attempt to fix the problem. However, the result was not satisfactory, because the devices were still opened by `RAND_set_rand_method()` and closed right afterwards in `rand_pool_cleanup()`. That's why I introduced `rand_cleaning_up`. 

So strictly speaking this swapping is not necessary anymore, but conceptually it makes more sense to cleanup the entropy source after the random generator has been disabled and not before.